johnzon-core/src/main/java/org/apache/johnzon/core/JsonPointerImpl.java

```diff
@@ -339,8 +339,7 @@ private JsonValue getValue(JsonValue jsonValue, String referenceToken, int curre
 
             try {
                 JsonArray jsonArray = (JsonArray) jsonValue;
-                int arrayIndex = Integer.parseInt(referenceToken);
-                validateArraySize(jsonArray, arrayIndex, jsonArray.size());
```

Comment:
seems we loose that validation for number cases, no?

Reply:
The array size validation will be executed in the validateArraySize (https://github.com/apache/johnzon/blob/d1c6a2f62ef0bb5a798161799eae4602e20a233e/johnzon-core/src/main/java/org/apache/johnzon/core/JsonPointerImpl.java#L469-L472).